### PR TITLE
Feature event loop sleep 33

### LINKED ISSUE #33: fix the event_loop_run loop to not waste cpu time when there's no io
> in https://github.com/meshlink/meshlink/blob/125279ed468788f67d4a6dc64353382bc88b8a6d/src/event.c#L236
> right now, as far I understand, while the socket is readable or writable select won't block but the event_loop_run loop will continue to loop with no pause till there's some blocking io or the socket neither has free buffer to send nor anything to read.
> 
> this also draws the timeout rather useless since nothing can be sent anyway while it is blocked
> 
> reading up on the code the timeout is set to pass when either the minimum set callback timeout (e.g. the mesh->pinginterval for mtu detection) or the lowest utcp retransmit timeout determined by the idle_cb calling utcp_timeout is passed, so for scheduled events.
> well, while nothing can be sent it still might be good to notify if send failures are handled correct but I rather expect additional trouble
> 
> to fix it, when there's nothing to receive or send, pause for a short moment
> ideally wait for event notification of newly available data or send calls but that won't work with the polling

### PR DESCRIPTION
**Pull request by everHannes**
Fixes #33 
based on https://github.com/meshlink/meshlink/pull/41 and https://github.com/meshlink/meshlink/pull/38

note that this not necessarily fixes the catta cpu spikes, didn't do any testing on this

### CODE REVIEW

**Comment by everHannes** [comment]
seems to work as well

**Comment by a-krebs** [comment]
Why has the design changed from one queue per mesh to a global queue?

File: src/meshlink_queue.h [hunk 1]
```diff
@@ -90,9 +92,16 @@ static inline void *meshlink_queue_peek(meshlink_queue_t *queue) {
 }
 
 /**
+ * Internal function to check whether there's something in the queue.
+ */
+static inline bool meshlink_queue_empty(meshlink_queue_t *queue) {
+    return queue->head != NULL;
```
Comment (a-krebs) [inline src/meshlink_queue.h]: This should be `==`. Also this function appears to be unused.

**Comment by a-krebs** [comment]
> Why has the design changed from one queue per mesh to a global queue?

Talked with @everHannes. Will revert to one queue for each mesh. This will be done in a separate pull request.

**Comment by everHannes** [comment]
> > Why has the design changed from one queue per mesh to a global queue?
> 
> Talked with @everHannes. Will revert to one queue for each mesh. This will be done in a separate pull request.

see #44

**Pull request merged**